routing/payment_lifecycle.go

```diff
@@ -203,20 +203,38 @@ func (p *paymentLifecycle) resumePayment(ctx context.Context) ([32]byte,
 		return [32]byte{}, nil, err
 	}
 
-	// Get the payment status.
-	status := payment.GetStatus()
-
 	// exitWithErr is a helper closure that logs and returns an error.
-	exitWithErr := func(err error) ([32]byte, *route.Route, error) {
-		// Log an error with the latest payment status.
-		//
-		// NOTE: this `status` variable is reassigned in the loop
-		// below. We could also call `payment.GetStatus` here, but in a
-		// rare case when the critical log is triggered when using
-		// postgres as db backend, the `payment` could be nil, causing
-		// the payment fetching to return an error.
-		log.Errorf("Payment %v with status=%v failed: %v", p.identifier,
-			status, err)
+	exitWithErr := func(exitErr error) ([32]byte, *route.Route, error) {
```

Comment:
if we really wanna do this let's make these into a new method so we don't grow the `resumePayment`